portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +32,39 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> comments;
+
+  @Override
+  public void init() {
+    comments = new ArrayList<String>();
+    comments.add("Wow, this is a great website!");
+    comments.add("I enjoyed the pictures that you shared.");
+    comments.add("The tech that you use is awesome!");
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Malcolm!</h1>");
+    String commentsJson = (new Gson()).toJson(comments);
+    response.setContentType("application/json;");
+    response.getWriter().println(commentsJson);
+  }
+
+  @Override
+  protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    
+    long timeStamp = System.currentTimeMillis();
+    String authorName = request.getParameter("comment-author");
+    String commentContent = request.getParameter("comment-input");
+
+    
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("time", timeStamp);
+    commentEntity.setProperty("author", authorName);
+    commentEntity.setProperty("content", commentContent);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
Not familiar with this service, but it is it possible to initialize this in init? Or do you need to do this every time.

Reply:
Looking at many other code examples that seems to be convention to put it in the `doPut ` method  but in my case it is also possible to put it in `init`. Ill do that approach.